src/index.ts

```diff
@@ -0,0 +1,518 @@
+interface CollectionConstructor {
+	new(): Collection<unknown, unknown>;
+	new<K, V>(entries?: ReadonlyArray<readonly [K, V]> | null): Collection<K, V>;
+	new <K, V>(iterable: Iterable<readonly [K, V]>): Collection<K, V>;
+	readonly prototype: Collection<unknown, unknown>;
+	readonly [Symbol.species]: CollectionConstructor;
+}
+
+/**
+ * A Map with additional utility methods. This is used throughout discord.js rather than Arrays for anything that has
+ * an ID, for significantly improved performance and ease-of-use.
+ * @extends {Map}
+ */
+class Collection<K, V> extends Map<K, V> {
+	private _array!: V[] | null;
```

Comment:
Shouldn't those get a docstring too?

Reply:
Good catch! I'll bring the docstring from the current JS version

Reply:
They have a docstring in the constructor. No need to document them again